central/image/service/service_impl.go

```diff
@@ -462,8 +471,15 @@ func (s *serviceImpl) GetImageVulnerabilitiesInternal(ctx context.Context, reque
 	return internalScanRespFromImage(img), nil
 }
 
-func (s *serviceImpl) acquireScanSemaphore() error {
-	if err := s.internalScanSemaphore.Acquire(concurrency.AsContext(concurrency.Timeout(maxSemaphoreWaitTime)), 1); err != nil {
+func (s *serviceImpl) acquireScanSemaphore(ctx context.Context) error {
+	semaphoreCtx, cancel := context.WithTimeout(ctx, maxSemaphoreWaitTime)
+	defer cancel()
+	if err := s.internalScanSemaphore.Acquire(semaphoreCtx, 1); err != nil {
+		// If the context was canceled, we do not want to indicate the client to retry.
+		if errors.Is(err, context.Canceled) {
+			return errors.Wrap(err, "acquiring scan semaphore")
+		}
+
 		// Aborted indicates the operation was aborted, typically due to a concurrency
 		// issues.  Clients should retry by default on Aborted.
 		s, err := status.New(codes.Aborted, err.Error()).WithDetails(&v1.ScanImageInternalResponseDetails_TooManyParallelScans{})
```

Comment:
Done outside this PR, but I'm trying to understand what happens when this returns `err != nil`. Seems like we just log it in prod builds and then return `nil` in the function? I don't think we want to do that, right?

Reply:
Ah good catch. I don't think we need the log anymore, and we should actually return the error to make sure we abort the caller function. Looking at the gRPC codes, it sounds like this would fit `INTERAL`:
> Internal errors. This means that some invariants expected by the underlying system have been broken. This error code is reserved for serious errors.